include/nlohmann/detail/macro_scope.hpp

```diff
@@ -132,12 +132,14 @@
              class NumberUnsignedType, class NumberFloatType,              \
              template<typename> class AllocatorType,                       \
              template<typename, typename = void> class JSONSerializer,     \
-             class BinaryType>
+             class BinaryType,                                             \
+             template<typename, typename> class LexerType,                 \
```

Comment:
I am not convinced that it is a good idea to template the type of the lexer and serializer. These are aspects of the parser/serializer, but not of the JSON type.

Reply:
I think I understand, but let me know. Would it make more sense to template the class(es) directly (`JSONNumerizer`)? Or even put it all into `JSONSerializer`?

Reply:
What I meant: The type of the lexer/serializer should not be a parameter of the JSON type. The JSON type should be defined by how values are stored. In the long term, a better library design could have the lexer/parser and serializer as free functions working on the public API of the JSON type. Then it would be easy to change these implementations without touching the JSON type in the first place.